### PR TITLE
Added offset argument to allow for nested progress bars

### PR DESCRIPTION
**Pull request by IsaacTay**
Adding feature from #52

### CODE REVIEW

File: src/ProgressMeter.jl [hunk 1]
```diff
@@ -111,40 +114,52 @@ mutable struct ProgressThresh{T<:Real} <: AbstractProgress
...
+    p.offset = offset
     t = time()
     if p.counter >= p.n
         if p.counter == p.n && p.printed
             percentage_complete = 100.0 * p.counter / p.n
             bar = barstring(p.barlen, percentage_complete, barglyphs=p.barglyphs)
             dur = durationstring(t-p.tfirst)
             msg = @sprintf "%s%3u%%%s Time: %s" p.desc round(Int, percentage_complete) bar dur
+            print(p.output, "\n" ^ p.offset)
+            print(p.output, "\n" ^ p.numprintedvalues)
```
Comment (timholy) [inline src/ProgressMeter.jl]: Combine these into `print(p.output, "\n"^(p.offset + p.numprintedvalues)` (it's approximately twice as fast, and easier to see the logic). This also applies several places below.

**Comment by IsaacTay** [comment]
Do you think it would be a better idea to implement it as a wrapper to AbstractProgress? It would allow for more flexibility, but it does require the use of an IOBuffer, which would slow it down quite a bit.

**Comment by IsaacTay** [comment]
[asciicast](https://asciinema.org/a/OopNqvmpHRLqpJyRsce2DSIiI)

**Comment by timholy** [comment]
I really like this and am ready to merge this. (Let's not do the wrapper idea for now, given the performance issue.)

@juliohm and @grero, can you please test this and see if it fits your needs?

**Comment by timholy** [comment]
CC also @ChrisRackaukas who gave the idea a thumbs-up.

**Comment by ChrisRackauckas** [comment]
This looks fantastic.

**Comment by juliohm** [comment]
Beautiful! :heart:

**Comment by grero** [comment]
This is really cool and exactly what I was looking for. Thanks!

**Comment by timholy** [comment]
@IsaacTay you have many fans, I hope to see you around the Julia community in the future!

Tagged release 0.8, when merged this will be broadly available.

**Pull request merged**